airflow/executors/celery_executor.py

```diff
@@ -104,8 +104,8 @@ def sync(self):
                         del self.tasks[key]
                         del self.last_state[key]
                     else:
-                        self.log.info("Unexpected state: %s", task.state)
-                    self.last_state[key] = task.state
+                        self.log.info("Unexpected state: %s", state)
+                        self.last_state[key] = state
```

Comment:
- The indentation change is to fix `potential bug-1`.

- Changing from `task.state` to `state` in both lines 107 and 108 are to fix `potential bug-2`